recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -23,22 +24,33 @@ this.EXPORTED_SYMBOLS = ["RecipeRunner"];
 
 const log = LogManager.getLogger("recipe-runner");
 const prefs = Services.prefs.getBranch("extensions.shield-recipe-client.");
+const TIMER_NAME = "recipe-client-addon-run";
 
 this.RecipeRunner = {
   init() {
     if (!this.checkPrefs()) {
       return;
     }
 
-    let delay;
     if (prefs.getBoolPref("dev_mode")) {
-      delay = 0;
-    } else {
-      // startup delay is in seconds
-      delay = prefs.getIntPref("startup_delay_seconds") * 1000;
+      // Run right now in dev mode
+      this.run();
     }
 
-    setTimeout(this.start.bind(this), delay);
+    // Run once every `runInterval` wall-clock seconds.
+    // This is managed by setting a "last ran" timestamp, and running if it is
+    // more than `runInterval` seconds ago.
```

Comment:
Does this mean that there's some minimum delay from the timer system we're using? When I tested locally and set the timeout to 5 seconds, it didn't repeat every 5 seconds, but at seemingly semi-random but frequent intervals.